.github/workflows/module-lint.yml

```diff
@@ -0,0 +1,69 @@
+name: Module Lint
+
+# runs at minute 10 of the 1st hour of every Sunday
+on:
+  schedule:
+    - cron: '10 1 * * 0'
+
+jobs:
+  prepare:
+    name: Prepare
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - name: Install Yarn dependencies
+        run: yarn --immutable
+
+  module-lint:
+    name: Module Lint
+    runs-on: ubuntu-latest
+    needs:
+      - prepare
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x, 20.x]
+        projects:
+          [
+            { name: 'auto-changelog', team: 'S051YKC9350' },
+            { name: 'create-release-branch', team: 'S051YKC9350' },
+            { name: 'module-lint', team: 'S051YKC9350' },
+            { name: 'utils', team: 'S051YKC9350' },
+          ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'yarn'
+      - run: yarn --immutable --immutable-cache
+      - run: yarn run-tool ${{ matrix.projects.name }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Require clean working directory
+        shell: bash
+        run: |
+          if ! git diff --exit-code; then
+            echo "Working tree dirty at end of job"
+            exit 1
+          fi
+      - name: Post module lint report to a slack
+        id: slack
+        if: ${{ failure() }}
```

Comment:
It looks like this step will perform if this job fails in any way. For instance, if the "Require clean working directory" step fails, then this step will run. We may not want to do that, but instead we may only want to do this only if `module-lint` fails to run.

What do you think about this?

``` suggestion
      - name: Run module-lint
        id: module-lint
        run: yarn run-tool ${{ matrix.projects.name }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      - name: Require clean working directory
        shell: bash
        run: |
          if ! git diff --exit-code; then
            echo "Working tree dirty at end of job"
            exit 1
          fi
      - name: Post module lint report to a slack
        id: slack
        if: ${{ failure() && steps.module-lint.outcome == 'failure' }}